src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/DynamicILGenerator.cs

```diff
@@ -475,6 +480,23 @@ private int GetMemberRefToken(MethodBase methodInfo, Type[]? optionalParameterTy
             return sig;
         }
 
+        private void AddParameters(SignatureHelper sigHelp, Type[]? parameterTypes, Type[][]? requiredCustomModifiers, Type[][]? optionalCustomModifiers)
+        {
+            if (requiredCustomModifiers != null && (parameterTypes == null || requiredCustomModifiers.Length != parameterTypes.Length))
```

Comment:
`parameterTypes == null` is treated as an empty array later, and we will ignore the `requiredCustomModifiers` in that case.

Should this condition be more like `if ((requiredCustomModifiers != null) && parameterTypes?.Length ?? 0 != requiredCustomModifiers.Length)` ?

I see that you have copied this over from the non-Dynamic case, so it may be a bug in that code too.

Reply:
The check disallow cases where `requiredcustommodifiers` is not null but `parameterTypes `is null. It throws if that is the case

Reply:
Ah ok